snap/local/pebble-wrapper

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh -eu
+
+# The default Pebble directory should sit within the $SNAP_USER_DATA because:
+# - access to the system's /var/lib/pebble/default isn't guaranteed, and
+# - the default $PEBBLE path should also be removed when removing the snap.
+DEFAULT_PEBBLE_DIRECTORY="${SNAP_USER_DATA}/var/lib/pebble/default"
+# We don't create this directory because the user can execute Pebble
+# with --create-dirs or even choose a different directory.
+export PEBBLE="${PEBBLE:-$DEFAULT_PEBBLE_DIRECTORY}"
```

Comment:
I don't know too much about snaps and wasn't part of the previous discussion, but presumably we have to do it with this wrapper script instead of using `environment` in the `snapcraft.yaml` so that we can set `PEBBLE` but only if it hasn't already been set by the user?

Reply:
Yes that's it. The problem with the `app.<name>.environment` is that (very much like a Pebble service) it will not be a default, but rather enforce an environment variable for the app. So you wouldn't be able to change it via the parent's environment.